File: TrippieBackend/TrippieBackend/Models/DTOs/Members/TripMemberDto.cs
using TrippieBackend.Models.Enums;

namespace TrippieBackend.Models.DTOs.Members;

public class TripMemberDto
{
    public Guid UserId { get; init; }
    
    public string Firstname { get; init; }
    
    public string Lastname { get; init; }
    
    public string Email { get; init; }
    
    public TripRoleEnum TripRole { get; init; }
    
    public DateTime JoinedAt { get; init; }
}

File: trippie_frontend/lib/features/trip/data/trip_member_dto.dart
class TripMemberDto {
  const TripMemberDto({
    required this.userId,
    required this.firstname,
    required this.lastname,
    required this.email,
    required this.tripRole,
    required this.joinedAt,
  });

  final String userId;
  final String firstname;
  final String lastname;
  final String email;
  final String tripRole;
  final DateTime joinedAt;

  factory TripMemberDto.fromJson(Map<String, dynamic> json) {
    return TripMemberDto(
      userId: json['userId'] as String,
      firstname: json['firstname'] as String,
      lastname: json['lastname'] as String,
      email: json['email'] as String,
      tripRole: json['tripRole'] as String,
      joinedAt: DateTime.parse(json['joinedAt'] as String),
    );
  }
}